### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by petre-c**
@Filiprogrammer, please note - https://github.com/Filiprogrammer/SimpleJadePinServer/issues/4 issue stands on debian as well as on docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+WORKDIR /app
+
+# Copy necessary files
+COPY . /app
+
+# Install dependencies
+RUN pip install wallycore
+
+# Generate a self-signed certificate (optional)
+RUN openssl req -new -x509 -keyout server.pem -out server.pem -days 3650 -nodes -subj "/CN=localhost"
```
Comment (Filiprogrammer) [inline Dockerfile:14]: Consider a bind mount or Docker secret instead of embedding the certificate directly into the container.

**Review by Filiprogrammer — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (petre-c) [reply Dockerfile:14]: Done

Comment (petre-c) [reply Dockerfile:14]: Actually, why not get rid of the need to use a self-signed certificate at all?

If protection becomes necessary, we could use a reverse-proxy, like nginx.

I tested removing the following line and it still seemed to work
```python
server.socket = ssl.wrap_socket(server.socket, certfile='./key_data/server.pem', server_side=True)
```

Comment (petre-c) [reply Dockerfile:14]: @Filiprogrammer can you please comment on this?

Comment (Filiprogrammer) [reply Dockerfile:14]: Once you get to step 3/4 and you click "Request Camera Permission" you will get the following error message when not using HTTPS on a non-localhost connection:
"Camera access is only supported in secure context like https or localhost."

This is the reason for the TLS certificate. The camera cannot be used without it.
As you mentioned one could use a reverse-proxy, so I just added a `--no-tls` command line option to disable TLS.

Comment (petre-c) [reply Dockerfile:14]: Understood, thank you for the clarification.

Is there anything that I need to address for you to accept this pull request?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Start with Python base image
+FROM python:3.9-slim
```
Comment (Filiprogrammer) [inline Dockerfile:2]:
```suggestion
FROM python:3.12-alpine
```
Switch to a more recent version of Python.
3.12 is the latest one supported by wallycore.
Also Python images of the alpine variant are about half the size of the slim images.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Set the working directory
+WORKDIR /app
+
+# Copy necessary files
+COPY . /app
+
+# Install dependencies
+RUN pip install wallycore
```
Comment (Filiprogrammer) [inline Dockerfile:11]:
```suggestion
RUN pip install --no-cache-dir wallycore
```
Do not use cache when installing packages to decrease the size of the resulting image.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Start with Python base image
+FROM python:3.9-slim
+
+# Set the working directory
+WORKDIR /app
+
+# Copy necessary files
+COPY . /app
```
Comment (Filiprogrammer) [inline Dockerfile:8]:
```suggestion
COPY SimpleJadePinServer.py index.html oracle_qr.html qrcode.js /app/
```
Only copy the necessary files to the container.

File: README.md [hunk 5]
```diff
@@ -20,13 +20,22 @@ pip install wallycore
 
 Or generate a self signed certificate
 
+
```
Comment (Filiprogrammer) [inline README.md:23]:
```suggestion
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+services:
+  simple-jade-pin-server:
+    build:
+      context: .
```
Comment (Filiprogrammer) [inline docker-compose.yml:4]:
```suggestion
    build: .
```
This can be abbreviated

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+    ports:
+      - "18080:4443"
+    volumes:
+      - ./key_data:/app/key_data
+    command: ["--no-tls"]
```
Comment (Filiprogrammer) [inline docker-compose.yml:9]: Nit: Add a new line at the end of a file

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+services:
+  simple-jade-pin-server:
```
Comment (Filiprogrammer) [inline docker-compose.yml:2]:
```suggestion
  simplejadepinserver:
```
Avoid using `-` in YAML keys due to potential compatibility problems.

File: Dockerfile (hunk 1 above)
Comment (Filiprogrammer) [reply Dockerfile:14]: A default certificate is still stored inside the Docker image, which is generally considered bad practice.
I believe it should be on the user to generate the certificate manually if they need it. _(Or use reverse proxy like Caddy with automatic certificate generation.)_

**Review by Filiprogrammer — CHANGES_REQUESTED** [review]
Thanks for the contributions so far!
I played around a bit with the Docker image and I still have some suggestions.

File: Dockerfile (hunk 1 above)
Comment (petre-c) [reply Dockerfile]: > I believe it should be on the user to generate the certificate manually if they need it.

Sure, agreed.

I removed the certificate generation inside the Docker image altogether as I'm passing `--no-tls` in `docker-compose.yml`

File: docker-compose.yml (hunk 8 above)
Comment (petre-c) [reply docker-compose.yml]: Changed from `-` to `_`

**Comment by petre-c** [comment]
I appreciate your level of attention to detail!

**Review by Filiprogrammer — APPROVED** [review]

**Pull request merged**